src/App.vue

```diff
@@ -1,11 +1,9 @@
-<script setup>
-import HomePage from "./pages/HomePage.vue";
-</script>
+<script setup></script>
 
 <template>
-  <HomePage />
+  <router-view name="HeaderPrimary"></router-view>
+  <router-view name="HeaderSecondary"></router-view>
```

Comment:
You should actually have only one named router-view here, let's say `<router-view name="header" />`

This way then in our router we could define an example page as:

```js
  {
    path: "/account",
    name: "UserAccount",
    components: {
      default: () => import("@/pages/UserAccount.vue"),
      header: () => import("@/components/header/HeaderPrimary.vue"),
    },
  },
  ```

So now we know there is **some** kind of a header, but which one - the routes decide 😄